modin/utils.py

```diff
@@ -137,17 +167,25 @@ def _inherit_docstrings(parent, excluded=[], overwrite_existing=False, apilink=N
 
     def _documentable_obj(obj):
         """Check if `obj` docstring could be patched."""
-        return callable(obj) or (isinstance(obj, property) and obj.fget)
+        return (
+            callable(obj)
+            or (isinstance(obj, property) and obj.fget)
+            or (isinstance(obj, (staticmethod, classmethod)) and obj.__func__)
+        )
 
     def decorator(cls_or_func):
         if parent not in excluded:
             _replace_doc(parent, cls_or_func, overwrite_existing, apilink)
 
         if not isinstance(cls_or_func, types.FunctionType):
-            for base in cls_or_func.__bases__:
+            seen = set()
+            for base in cls_or_func.__mro__:
```

Comment:
I would highlight in the docs of `_inherit_docstrings` that it may overwrite docstrings of the base classes of the decorated object. It's obvious when looking into the code, but not obvious at all when you're reading the function's contract.

```python
from modin.utils import _inherit_docstrings

class DocsContainer:
    def method():
        """Doc-string."""
        pass

class Base:
    def method(): # Not documented method
        pass

@_inherit_docstrings(DocsContainer)
class Derived(Base):
    pass

print(Derived.method.__doc__) # Doc-string.
print(Base.method.__doc__)    # Doc-string.

```

Reply:
Good point!